server/src/jobs/hydrate/organismes/hydrate-organismes.js

```diff
@@ -0,0 +1,156 @@
+import logger from "../../../common/logger.js";
+import {
+  createOrganisme,
+  findOrganismeByUaiAndSiret,
+  updateOrganisme,
+} from "../../../common/actions/organismes/organismes.actions.js";
+import { createJobEvent } from "../../../common/actions/jobEvents.actions.js";
+import { organismesDb, organismesReferentielDb } from "../../../common/model/collections.js";
+import { PromisePool } from "@supercharge/promise-pool/dist/promise-pool.js";
+
+const JOB_NAME = "hydrate-organismes";
+
+let nbOrganismeCreated = 0;
+let nbOrganismeNotCreated = 0;
+let nbOrganismeUpdated = 0;
+let nbOrganismeNotUpdated = 0;
+
+/**
+ * Script qui initialise le stock des organismes
+ * 1. On va créer tous les organismes "stock" non présents dans le tdb mais existants dans le référentiel
+ * 2. Pour les organismes déja présents va MAJ les organismes présent
+ * Le format adresse des organismes du référentiel est différent du format tdb donc on va transformer le champ adresse
+ * En cas d'erreurs on log via un createJobEvent()
+ */
+export const hydrateOrganismesFromReferentiel = async () => {
+  // On remet à 0 l'information de présence dans le référentiel
+  await resetOrganismesReferentielPresence();
+
+  // On récupère l'intégralité des organismes depuis le référentiel
+  const organismesFromReferentiel = await organismesReferentielDb().find().toArray();
+  logger.info(`Traitement de ${organismesFromReferentiel.length} organismes provenant du référentiel...`);
+
+  // Processes 10 organismes en // par défaut
+  await PromisePool.for(organismesFromReferentiel).process(async (currentOrganismeFromReferentiel) => {
```

Comment:
pas bloquant mais plus concis: `await PromisePool.for(organismesFromReferentiel).process.for(insertOrUpdateOrganisme)`

Reply:
👉🏼 https://github.com/mission-apprentissage/flux-retour-cfas/pull/2599/commits/23e7d70b46fb592fef54361b42b27a6e928459ea